test/Microsoft.Extensions.Primitives.Tests/CompositeChangeTokenTest.cs

```diff
@@ -0,0 +1,175 @@
+
+            // Act & Assert
+            Assert.True(compositeChangeToken.HasChanged);
+        }
+
+        [Fact]
+        public void HasChanged_IsFalse_IfNoTokenHasChanged()
+        {
+            // Arrange
+            var firstChangeToken = new MockChangeToken();
+            var secondChangeToken = new MockChangeToken();
+
+            var compositeChangeToken = new CompositeChangeToken(new List<IChangeToken> { firstChangeToken, secondChangeToken });
+
+            // Act & Assert
+            Assert.False(compositeChangeToken.HasChanged);
+        }
+
+        [Fact]
+        public void ActiveChangeCallbacks_IsTrue_IfAnyTokenHasActiveChangeCallbacks()
+        {
+            // Arrange
+            var firstChangeToken = new MockChangeToken();
+            var secondChangeToken = new MockChangeToken { ActiveChangeCallbacks = true };
+            var thirdChangeToken = new MockChangeToken();
+
+            var compositeChangeToken = new CompositeChangeToken(new List<IChangeToken> { firstChangeToken, secondChangeToken, thirdChangeToken });
+
+            // Act & Assert
+            Assert.True(compositeChangeToken.ActiveChangeCallbacks);
+        }
+
+        [Fact]
+        public void ActiveChangeCallbacks_IsFalse_IfNoTokenHasActiveChangeCallbacks()
+        {
+            // Arrange
+            var firstChangeToken = new MockChangeToken();
+            var secondChangeToken = new MockChangeToken();
+
+            var compositeChangeToken = new CompositeChangeToken(new List<IChangeToken> { firstChangeToken, secondChangeToken });
+
+            // Act & Assert
+            Assert.False(compositeChangeToken.ActiveChangeCallbacks);
+        }
+
+        [Fact]
+        public async Task CallbackRaisedOnce_ConcurrentThreadsBlocked()
+        {
+            // Arrange
+            var event1 = new AutoResetEvent(false);
+            var event2 = new AutoResetEvent(false);
+            var event3 = new AutoResetEvent(false);
+
+            var firstChangeToken = new MockChangeToken();
+            var secondChangeToken = new MockChangeToken() { ActiveChangeCallbacks = true };
+            var count = 0;
+            Action<object> callback = _ =>
+            {
+                count++;
+                event3.Set();
+                event1.WaitOne(5000);
+            };
+
+            var compositeChangeToken = new CompositeChangeToken(new List<IChangeToken> { firstChangeToken, secondChangeToken });
+            compositeChangeToken.RegisterChangeCallback(callback, null);
+
+            // Act
+            var firstChange = Task.Run(() =>
+            {
+                event2.WaitOne(5000);
+                firstChangeToken.RaiseCallback(null);
+            });
+            var secondChange = Task.Run(() =>
+            {
+                event3.WaitOne(5000);
+                secondChangeToken.RaiseCallback(null);
+            }).ContinueWith(x => event1.Set());
```

Comment:
`event1` is what pauses the first thread while the other makes its way to the same callback. In theory, the second thread is not allowed in the callback and after trying to RaiseCallback, `ContinueWith(x=> event1.Set())` ensures that event1 is released.